src/lib/io/sol-i2c-impl-linux.c

```diff
@@ -968,8 +968,22 @@ create_device_iter_cb(void *data, const char *dir_path, struct dirent *ent)
     char path[PATH_MAX];
     int r;
     struct stat st;
+#ifdef USE_I2C
```

Comment:
I'd say that this file is only compiled when using I2C, so this ifdef is useless.